lib/src/element_type.dart

```diff
@@ -23,7 +23,11 @@ class ElementType {
 
   bool get isParameterizedType {
     if (_type is FunctionType) {
-      return (_type as FunctionType).typeFormals.isNotEmpty;
+      final FunctionType functionType = _type;
+      final Iterable<DartType> typeArguments = []
+          ..addAll(functionType.typeArguments)
+          ..addAll(functionType.typeFormals.map((f) => f.type));
+      return typeArguments.isNotEmpty;
     } else if (_type is ParameterizedType) {
```

Comment:
Wow, thanks a lot for the extended answer, advice and the code snippet! I used it as is, and it works correctly. Thanks!